### ISSUE TITLE
Joy message wasn't getting the extra axis from knuckle controller 

### ISSUE THREAD

**Issue by zp-yang** [asymingt/libsurvive_ros2#11]
Knuckle controllers have 11 axes as input, but the original max count was 8 so the pad force, stick-x and stick-y weren't mapped to the joy message. Submitted a simple fix for this by doubling the max count. #10
![knuckle-buttons](https://github.com/asymingt/libsurvive_ros2/assets/39218136/ff5c9f7c-357c-4b42-b535-7a909e528ba2)

**Comment by asymingt** [comment]
Is doubling the capacity needed because there are two controllers appearing as one device, or was that just chosen randomly to add capacity so as not to overflow?

**Comment by zp-yang** [comment]
good point, I originally just chose to double it so as to not overflow, but
libsurvive's survive-buttons.c
https://github.com/cntools/libsurvive/blob/037189ebbcdb2a0bedd6fe6a8d1ad3910ac6caad/survive-buttons.c#L115
seems to loop over 16 axes instead of the max count of 8 in their api. Not
sure if the fix should be in the survive api or this ros2 package. What do
you think?

**Comment by asymingt** [comment]
I think we should match whatever they do. I think it must have been my error when converting from their ROS1 to a ROS2 node. Are you happy to update the loop to span 16 elements?

**Comment by zp-yang** [comment]
> I think we should match whatever they do. I think it must have been my error when converting from their ROS1 to a ROS2 node. Are you happy to update the loop to span 16 elements?

Sure thing. Since SURVIVE_MAX_AXIS_COUNT is already 8, doubling it would be 16 and that is what my hacky fix was. Or do your prefer to hard code a '16' for the loop count?

**Comment by asymingt** [comment]
It's not a big deal. To me the doubling gives the sense that up to two logical devices are supported, whereas it seems more like that there are just more values that need to be packed. So, why not just #define a new max of 16 and use the descriptive variable name? I can do that if you're busy. Either way, thank you very much for finding and fixing this bug!
